### DISCUSSION TITLE
Overfitting in Radial Velocity Data + Parameter Issues

### DISCUSSION THREAD

**Discussion by AjlaTr** [phoebe-project/phoebe2#848] [Questions]
I'm working with SB1 binaries and using PHOEBE's rv_geom estimator to extract parameters like period and eccentricity. However, when trying to synthesize more radial velocity data, the model drastically overfits my observations. I attempted to correct this by initializing parameters but encountered difficulties in solving the inverse problem. The error I got along with the graph is included below. My work revolves around the idea that the secondary component is a noninteracting black hole, so there will be no observational data on it. It also may be important to note that the data on the visible star is not continuous. Could you advise on effectively synthesizing data without overfitting and ensuring an accurate model representation given these challenges?
![K2285PHOEBE](https://github.com/phoebe-project/phoebe2/assets/167161028/8c20aaab-8e0f-4875-94be-fb20b07fcda5)
![Screenshot 2024-04-16 215659](https://github.com/phoebe-project/phoebe2/assets/167161028/b7fea335-51f7-4e2f-a60a-35ff46be5f2e)

**Comment by kecnry** [comment]
Can you show this plot in phase-space, by passing `x='phases'` to the plot command?

**Reply by AjlaTr** [reply]
Of course. Here is the plot:
![Screenshot 2024-04-17 102506](https://github.com/phoebe-project/phoebe2/assets/167161028/23ebec58-5d71-4c8f-85f7-b9d48dac8450)
Thanks for your quick reply!

**Reply by kecnry** [reply]
ok, is your orbital period already set?  The rv geometry estimator requires phased RV data as input (which requires a known/estimated orbital period).  Once you have that, you can get around the error you are seeing (I'm guessing it was raised during a call to `adopt_solution()`?) by following the advice in the error message and choosing which of those parameters should "absorb" the _proposed_ changes to asini.  For a general background on constraints, see [this tutorial](http://phoebe-project.org/docs/latest/tutorials/constraints).

**Reply by AjlaTr** [reply]
The orbital period is unset because we hoped to extract both period and eccentricity from the model. We are looking for these values in order to apply the mass function to the SB1 binary. While we have certain information on the star (effective temperature, stellar mass) there is no data on the unseen companion. We are currently working with radial velocity time series. I included a snippet of the data we are using. Would this be suitable to extract the specific parameters I mentioned above with PHOEBE?

[K2285dataSnippet - K2285data.csv](https://github.com/phoebe-project/phoebe2/files/15017798/K2285dataSnippet.-.K2285data.csv)

**Reply by kecnry** [reply]
My suggestion would be to at least estimate the value of the orbital period as the first step so that you can phase-fold the data to estimate the amplitude and ultimately get a physical model that is close to representing the data.  At that point, you can always allow the period to be revised when optimizing/sampling.

It's hard to tell without seeing the phased data if eccentricity will be well-constrained or not, but in theory, if the RV is well-sampled in phase-space, you should be able to get an eccentricity.

**Reply by AjlaTr** [reply]
I will try that out to see if it works. Thanks again for all your help!
Also, I was wondering if you knew of any other PHOEBE library that would work for this besides the RV geometry estimator.

**Reply by kecnry** [reply]
The RV geometry estimator just measures and calculates _estimates_ for these parameters, and is just the first step in fitting an RV to the data.  In your case, an analytic RV (outside phoebe entirely) is probably going to be significantly faster to fit than the numerical RV within phoebe.

**Reply by AjlaTr** [reply]
I will certainly look into that. Thank you!
